easybuild/tools/github.py

```diff
@@ -718,12 +761,181 @@ def update_pr(pr, paths, commit_msg=None):
+
+    # try to clone repo and push a test branch
+    git_working_dir = tempfile.mkdtemp(prefix='git-working-dir')
+    git_repo, res = None, None
+    branch_name = 'test_branch_%s' % ''.join(random.choice(string.letters) for _ in range(5))
+    try:
+        git_repo = init_repo(git_working_dir, GITHUB_EASYCONFIGS_REPO, silent=True)
+        setup_repo(git_repo, github_user, GITHUB_EASYCONFIGS_REPO, 'master', silent=True)
+        git_repo.create_head(branch_name)
+        res = git_repo.remotes.origin.push(branch_name)
+    except Exception as err:
+        _log.warning("Exception when testing push access to %s/%s: %s", github_user, GITHUB_EASYCONFIGS_REPO, err)
+
+    if res:
+        if res[0].flags & res[0].ERROR:
+            _log.warning("Error occured when pushing test branch to GitHub: %s", res[0].summary)
+            check_res = "FAIL (error occured)"
+        else:
+            check_res = "OK"
+    else:
+        check_res = "FAIL (unexpected exception)"
+
+    if 'FAIL' in check_res:
+        status['--new-pr'] = status['--update-pr'] = False
+
+    print check_res
+
+    # cleanup: delete test branch that was pushed to GitHub
+    if git_repo:
+        try:
+            git_repo.remotes.origin.push(branch_name, delete=True)
+        except GitCommandError:
+            pass
+
+    # test creating a gist
+    print "* creating gists...",
+    res = None
+    try:
+        res = create_gist("This is just a test", 'test.txt', descr='test123', github_user=github_user)
+    except Exception as err:
+        _log.warning("Exception occured when trying to create gist: %s", err)
+
+    if res and re.match('https://gist.github.com/[0-9a-f]+$', res):
+        check_res = "OK"
+    else:
+        check_res = "FAIL (res: %s)" % res
+        status['--upload-test-report'] = False
+
+    print check_res
+
+    # check whether location to local working directories for Git repositories is available (not strictly needed)
+    print "* location to Git working dirs...",
+    git_working_dirs_path = build_option('git_working_dirs_path')
+    if git_working_dirs_path:
+        print "OK (%s)" % git_working_dirs_path
+    else:
+        print "not found (suboptimal)"
+
+    # report back
+    if all(status.values()):
+        print "\nAll checks PASSed!\n"
+    else:
+        print '\n'.join([
+            '',
+            "One or more checks FAILed, GitHub configuration not fully complete!",
+            "See http://easybuild.readthedocs.org/en/latest/Integration_with_GitHub.html#configuration for help.",
+            '',
+        ])
+
+    print "Status of GitHub integration:"
+    for action in sorted(status):
+        print "* %s:" % action,
+        if status[action]:
+            print "OK"
+        else:
+            print "not supported"
+    print ''
+
+
 class GithubToken(object):
     """Representation of a GitHub token."""
 
-    # singleton metaclass: only one instance is created
-    __metaclass__ = Singleton
```

Comment:
There shouldn't be, there can only be one token available for a particular user anyway

It has a slight performance impact in case the token is fetched multiple times, but that's negligible.

if anything, it has advantages, certainly for testing...